core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -578,14 +578,17 @@ class ReplicaManager(val config: KafkaConfig,
           if (!logManager.isLogDirOnline(destinationDir))
             throw new KafkaStorageException(s"Log directory $destinationDir is offline")
 
-          // Stop current replica movement if the destinationDir is different from the existing destination log directory
-          getReplica(topicPartition, Request.FutureLocalReplicaId) match {
-            case Some(futureReplica) =>
-              if (futureReplica.log.get.dir.getParent != destinationDir) {
+          getPartition(topicPartition) match {
+            case Some(partition) =>
+              if (partition eq ReplicaManager.OfflinePartition)
+                throw new KafkaStorageException()
```

Comment:
Is there any kind of useful message we can add here?

Reply:
The exception will be caught in the same method `alterReplicaLogDirs()` and the message will be lost when it is translated by `Errors.forException(e)`. Now I realized that the code may be more readable by having a message. I added `s"Partition $topicPartition is offline"` now. Thanks.